experimental/dds/tree2/src/feature-libraries/sequence-field/rebase.ts

```diff
@@ -1002,16 +1006,26 @@ function compareCellPositions(
 		}
 	}
 
-	if (isNewAttach(newMark)) {
-		// When the marks are at the same position, we use the tiebreak of `newMark`.
-		// TODO: Use specified tiebreak instead of always tiebreaking left.
+	if (newId.revision === undefined) {
+		// An undefined revision must mean that the cell was created on the branch we are rebasing.
+		// Since it is newer than the `baseMark`'s cell, it should come first.
 		return Infinity;
 	}
 
-	assert(
-		isNewAttach(baseMark),
-		0x6a1 /* Lineage should determine order of marks unless one is a new attach */,
-	);
+	const baseRevisionIndex = metadata.getIndex(baseId.revision);
+	const newRevisionIndex = metadata.getIndex(newId.revision);
+
+	if (newRevisionIndex !== undefined && baseRevisionIndex !== undefined) {
+		return newRevisionIndex > baseRevisionIndex ? Infinity : -Infinity;
+	}
+
+	if (newRevisionIndex !== undefined) {
+		return Infinity;
+	}
+
+	if (baseRevisionIndex !== undefined) {
+		return -Infinity;
+	}
 
 	// `newMark` points to cells which were emptied before `baseMark` was created.
```

Comment:
THhis whole comment block presumably need updating

Reply:
Actually, do we ever expect `baseRevisionIndex` to be undefined? What would that even mean?

Reply:
Sorry, got confused. It should occur if base is targetting a cell that was created before any of the base changes.